flytekit/core/interface.py

```diff
@@ -573,3 +574,30 @@ def remap_shared_output_descriptions(output_descriptions: Dict[str, str], output
         return output_descriptions
     _, shared_description = next(iter(output_descriptions.items()))
     return {k: shared_description for k, _ in outputs.items()}
+
+
+def get_documentation_from_docstring(
+    docstring: Optional[Docstring], default_val: Optional[Documentation] = None
+) -> Documentation:
+    """
+    Construct a Documentation object from the docstring and default value
+
+    :param docstring: the docstring from the task/workflow
+    :param default_val: default value
+    :return: documentation object
+    """
+    doc = default_val or Documentation()
+
+    if docstring is None:
+        return doc
+
+    if docstring.short_description:
+        doc.short_description = docstring.short_description
+    if docstring.long_description:
+        if sys.getsizeof(docstring.long_description) > 16 * 1024 * 1024:
+            # Check if the size of long description exceeds 16KB
+            raise ValueError(
+                "Long Description of the flyte entity exceeds the 16KB size limit. Please specify the uri in the long description instead."
+            )
```

Comment:
Erroring on a long docstring does not feel great. I'll rather slice the docstring with `textwrap.shorten` and add some context saying that "the docstring was too long"

Since this behavior is already in `master`, I am okay with leaving this as is in this PR.